src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -267,34 +265,64 @@ minRequiredLength switch
+                    EmitTextInfo(writer, ref hasTextInfo, rm);
+
+                    // Emit the code for whatever find mode has been determined.
+                    switch (code.FindOptimizations.FindMode)
                     {
-                        EmitLeadingCharacter_LeftToRight();
+                        case FindNextStartingPositionMode.LeadingPrefix_LeftToRight_CaseSensitive:
+                            Debug.Assert(!string.IsNullOrEmpty(code.FindOptimizations.LeadingCaseSensitivePrefix));
+                            EmitIndexOf_LeftToRight(code.FindOptimizations.LeadingCaseSensitivePrefix);
+                            break;
+
+                        case FindNextStartingPositionMode.LeadingPrefix_RightToLeft_CaseSensitive:
+                            Debug.Assert(!string.IsNullOrEmpty(code.FindOptimizations.LeadingCaseSensitivePrefix));
+                            EmitIndexOf_RightToLeft(code.FindOptimizations.LeadingCaseSensitivePrefix);
+                            break;
+
+                        case FindNextStartingPositionMode.FixedSets_LeftToRight_CaseSensitive:
+                        case FindNextStartingPositionMode.FixedSets_LeftToRight_CaseInsensitive:
+                        case FindNextStartingPositionMode.LeadingSet_LeftToRight_CaseSensitive:
+                        case FindNextStartingPositionMode.LeadingSet_LeftToRight_CaseInsensitive:
+                            Debug.Assert(code.FindOptimizations.FixedDistanceSets is { Count: > 0 });
+                            EmitFixedSet_LeftToRight();
+                            break;
+
+                        case FindNextStartingPositionMode.LeadingSet_RightToLeft_CaseSensitive:
+                        case FindNextStartingPositionMode.LeadingSet_RightToLeft_CaseInsensitive:
+                            Debug.Assert(code.FindOptimizations.FixedDistanceSets is { Count: > 0 });
+                            EmitFixedSet_RightToLeft();
+                            break;
+
+                        // Already emitted earlier
+                        case FindNextStartingPositionMode.LeadingAnchor_LeftToRight_Beginning:
+                        case FindNextStartingPositionMode.LeadingAnchor_LeftToRight_End:
+                        case FindNextStartingPositionMode.LeadingAnchor_LeftToRight_EndZ:
+                        case FindNextStartingPositionMode.LeadingAnchor_LeftToRight_Start:
+                        case FindNextStartingPositionMode.LeadingAnchor_RightToLeft_Beginning:
+                        case FindNextStartingPositionMode.LeadingAnchor_RightToLeft_End:
+                        case FindNextStartingPositionMode.LeadingAnchor_RightToLeft_EndZ:
+                        case FindNextStartingPositionMode.LeadingAnchor_RightToLeft_Start:
+                            Debug.Assert(code.FindOptimizations.LeadingAnchor != 0);
+                            goto case FindNextStartingPositionMode.NoSearch;
```

Comment:
Will this line ever be hit? If we don't expect to, should we instead throw an exception since we are in a weird state?

Reply:
Mostly not anymore.  There's one corner case it I'd left it in for, but it's so rare and not worth optimizing for that it's not worth cluttering up the code for, so I've deleted it and then all this.